ql/src/java/org/apache/hadoop/hive/ql/udf/generic/GenericUDTFGetSplits.java

```diff
@@ -559,12 +566,22 @@ private SplitResult getSplits(JobConf job, TezWork work, Schema schema, Applicat
         // 4. Make location hints.
         SplitLocationInfo[] locations = makeLocationHints(hints.get(i));
 
+        // 5. populate info about llap daemons(to help client submit request and read data)
+        LlapDaemonInfo[] llapDaemonInfos = populateLlapDaemonInfos(job, locations);
+
+        // 6. Generate JWT for external clients if it's a cloud deployment
+        String jwt = "";
+        if (LlapUtil.isCloudDeployment()) {
+          JwtHelper jwtHelper = new JwtHelper(SessionState.getSessionConf());
+          jwt = jwtHelper.buildJwtForLlap(applicationId);
```

Comment:
leave a comment about which application's id we are injecting here and how the flow of jwt validation works..

Reply:
done